modules/basic-skills/src/backend/choice_ndu.ts

```diff
@@ -0,0 +1,148 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import yn from 'yn'
+
+const generateFlowNdu = async (data: any): Promise<sdk.FlowGenerationResult> => {
+  const { randomId } = data
+
+  const hardRetryLimit = 10
+  const nbMaxRetries = Math.min(Number(data.config.nbMaxRetries), hardRetryLimit)
+  const repeatQuestion = yn(data.config.repeatChoicesOnInvalid)
+
+  const sorrySteps = []
+
+  if (data.invalidContentId && data.invalidContentId.length >= 3) {
+    sorrySteps.push({
+      type: sdk.NodeActionType.RenderElement,
+      name: `#!${data.invalidContentId}`
```

Comment:
Yup I agree, we could add a method to common to build it, like if it's an element, a content type, etc.. But that's the scope of another PR, since that's used almost everywhere there's a couple of places to change it & more tests